pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -480,6 +484,17 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 			if !isInternal {
 				service.Annotations[alibabaCloudLBAddressTypeAnnotation] = alibabaCloudLBAddressTypeInternet
 			}
+		case configv1.OpenStackPlatformType:
+			// Set a floating IP only if the load balancer scope is external.
+			if !isInternal {
+				if lbSpec != nil && lbSpec.ProviderParameters != nil &&
+					lbSpec.ProviderParameters.Type == operatorv1.OpenStackLoadBalancerProvider &&
+					lbSpec.ProviderParameters.OpenStack != nil {
+					// We know that LoadBalancerIP is deprecated but in this current version of Service, this won't be removed.
+					// It'll probably be removed in Service v2 which is not going to happen in the near future.
+					service.Spec.LoadBalancerIP = lbSpec.ProviderParameters.OpenStack.FloatingIP
```

Comment:
I went to check whether validation for `service.Spec.LoadBalancerIP` was stricter than validation for `lbSpec.ProviderParameters.OpenStack.FloatingIP` (that is to say, whether `FloatingIP` could have some value that would be invalid for `LoadBalancerIP`), but I cannot find any validation for `LoadBalancerIP` whatsoever!  But I suppose that's exactly the reason why it's being deprecated.  * grin *.  
https://github.com/openshift/cluster-ingress-operator/blob/80bf760ba715300c97fdc5d6188150125154097b/vendor/k8s.io/api/core/v1/types.go#L5421

Reply:
yeah not ideal :cry: